record-boot-success/record-boot-success

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/python3
+import dbus
+import dbus.service
+import logging
+import os
+import subprocess
+
+from systemd import journal
+
+BUSNAME = "com.endlessm.RecordBootSuccess"
+
+
+def record_boot_success():
+    envfile = "/boot/grub/grubenv"
+    program = "/usr/bin/grub-editenv"
+
+    try:
+        if os.path.getsize(program) == 0:
```

Comment:
What's the purpose of this? In what scenario would this file exist but be empty? Surely running it a few lines below would fail in that case?

Reply:
The shell script this is replacing is:

```
 [ -s /boot/grub/grubenv ] || rm -f /boot/grub/grubenv 
```

Note that it's checking for a zero-size `grubenv`, not a zero-size executable.

Reply:
Yep, that was a typo. And I don't have the answer for why `/boot/grub/grubenv` would be empty, but this comes from the original Debian init script that replaced on https://github.com/endlessm/grub/commit/e3a6fc3f37818f6bb169100017588679147ceacb.

Manually making `/boot/grub/grubenv` an empty file makes grub-editenv fail, and if we boot in that state GRUB prints an error message and boots fine, but it does not set the "recordfail" mark, so it is probably a good idea to keep checking for that.